daemon/info.go

```diff
@@ -119,7 +134,7 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 		MemTotal:           meminfo.MemTotal,
 		GenericResources:   daemon.genericResources,
 		DockerRootDir:      daemon.configStore.Root,
-		Labels:             daemon.configStore.Labels,
+		Labels:             daemon.getFilteredLabels(),
```

Comment:
Wondering if this filter should be applied when _reading_ the configuration, or when changing/updating the configuration (i.e., filter out user-provided labels with the prefix)

Reply:
Possibly - I was wondering the same.  I left it unchanged on the `config.Config` object, since that seems to be the configuration as specified by the user, but I wasn't sure.  I'd be happy to make the change if you feel that'd be a better place to do it, though.

Reply:
Yes; so train of thought is that there's possibly other locations where we use this information, and those may not take the filtering into account.

Perhaps we should validate when loading, and print a warning that the label will be ignored; in future we could turn that warning into an error

Reply:
Done